tap_ilevel/transform.py

```diff
@@ -1,42 +1,19 @@
-import re
-
-# Convert camelCase to snake_case
-def convert(name):
-    regsub = re.sub('(.)([A-Z][a-z]+)', r'\1_\2', name)
-    return re.sub('([a-z0-9])([A-Z])', r'\1_\2', regsub).lower()
-
-
-# Convert keys in json array
-def convert_array(arr):
-    new_arr = []
-    for i in arr:
-        if isinstance(i, list):
-            new_arr.append(convert_array(i))
-        elif isinstance(i, dict):
-            new_arr.append(convert_json(i))
-        else:
-            new_arr.append(i)
-    return new_arr
+#!/usr/bin/env python3
 
-
-# Convert keys in json
-def convert_json(this_json):
-    out = {}
-    for key in this_json:
-        new_key = convert(key)
-        if isinstance(this_json[key], dict):
-            out[new_key] = convert_json(this_json[key])
-        elif isinstance(this_json[key], list):
-            out[new_key] = convert_array(this_json[key])
-        else:
-            out[new_key] = this_json[key]
-    return out
+import re
+import hashlib
+import humps
 
 
 #  camelCase to snake_case for fieldname keys
 def transform_json(records):
-    records = {
-        'records': records
-    }
-    transformed_json = convert_json(records)
-    return transformed_json['records']
+    transformed_json = humps.decamelize(records)
+    return transformed_json
+
+
+# Create MD5 hash key for data element
+def hash_data(data):
```

Comment:
This function does not seem to be referenced anywhere?

Reply:
`hash_data` is imported and used in `sync.py` and `ilevel_api.py` to simplify a multi-dimension key.